src/filters/arrays.js

```diff
@@ -50,8 +51,9 @@ function matchItems(array1, array2, index1, index2, context) {
   }
   var objectHash = context.objectHash;
   if (!objectHash) {
-    // no way to match objects was provided, try match by position
-    return context.matchByPosition && index1 === index2;
+    var ivalue1 = Immutable.Map(value1);
+    var ivalue2 = Immutable.Map(value2);
+    return Immutable.is(ivalue1, ivalue2);
```

Comment:
is this doing a deep compare?
the problem with that is it would only match 2 objects if they're exactly equal, which is fine when an object just changed position, but if instead there was a nested small change, they wouldn't match, causing a full remove + add (instead of diffing inside), so I'm not sure I'd do a deep compare as default behavior.